main.tf

```diff
@@ -20,6 +20,7 @@ resource "azurerm_kubernetes_cluster" "main" {
   resource_group_name                 = data.azurerm_resource_group.main.name
   automatic_channel_upgrade           = var.automatic_channel_upgrade
   azure_policy_enabled                = var.azure_policy_enabled
+  cost_analysis_enabled               = var.cost_analysis_enabled
```

Comment:
According to [the document](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#cost_analysis_enabled):

>The sku_tier must be set to `Standard` or `Premium` to enable this feature. 

Could we add a `precondition` [here](https://github.com/Azure/terraform-azurerm-aks/blob/main/main.tf#L626)?

```hcl
precondition {
  condition         = var.cost_analysis_enabled != true || (var.sku_tier == "Standard" || var.sku_tier == "Premium")
  error_message = "`sku_tier` must be either `Standard` or `Premium` when cost analysis is enabled."
}
```